torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1697,7 +1697,10 @@ std::exception_ptr ProcessGroupNCCL::checkForNCCLErrorsInternal(
               *commFailureReason)));
     }
     ncclResult_t ncclAsyncErr = ncclComm->checkForNcclError();
-    if (ncclAsyncErr != ncclSuccess) {
+    // When nonblocking mode is enabled by TORCH_NCCL_USE_COMM_NONBLOCKING,
```

Comment:
is checkForNCCLErrors only called from the work processing loop or other places too?  this looks right to me (from the assumption of the work loop).

Reply:
`ncclInProgress` is usually observed by the main thread, before the CPU work of an API call actually finishes. Here, since the watchdog thread constantly runs on the side, it may happen to observe `ncclInProgress` too if it wakes up at the "right" time. I guess here we just mute the watchdog in case it thinks there is an error.